publisher-sdk-tests/src/androidTest/java/com/criteo/publisher/headerbidding/MoPubHeaderBiddingTest.java

```diff
@@ -68,6 +68,16 @@ public void canHandle_GivenMoPubView_ReturnTrue() throws Exception {
     assertTrue(handling);
   }
 
+  @Test
+  public void canHandle_GivenSubClassOfMoPubView_ReturnTrue() throws Exception {
+    MoPubView moPub = callOnMainThreadAndWait(() -> new MoPubView(activityRule.getActivity()) {
```

Comment:
Does this anonymous class really represent a subclass of `MoPubView`?

Reply:
Yep, this is the shorter way to declare a subclass in Java.

Here is the DEX bytecode to verify this:
```
.class Lcom/criteo/publisher/headerbidding/MoPubHeaderBiddingTest$1;
.super Lcom/mopub/mobileads/MoPubView;
.source "MoPubHeaderBiddingTest.java"


# annotations
.annotation system Ldalvik/annotation/EnclosingMethod;
    value = Lcom/criteo/publisher/headerbidding/MoPubHeaderBiddingTest;->lambda$canHandle_GivenSubClassOfMoPubView_ReturnTrue$0()Lcom/criteo/publisher/headerbidding/MoPubHeaderBiddingTest$1;
.end annotation

.annotation system Ldalvik/annotation/InnerClass;
    accessFlags = 0x0
    name = null
.end annotation


# instance fields
.field final synthetic this$0:Lcom/criteo/publisher/headerbidding/MoPubHeaderBiddingTest;


# direct methods
.method constructor <init>(Lcom/criteo/publisher/headerbidding/MoPubHeaderBiddingTest;Landroid/content/Context;)V
    .registers 3
    .param p1, "this$0"    # Lcom/criteo/publisher/headerbidding/MoPubHeaderBiddingTest;
    .param p2, "x0"    # Landroid/content/Context;

    .line 73
    iput-object p1, p0, Lcom/criteo/publisher/headerbidding/MoPubHeaderBiddingTest$1;->this$0:Lcom/criteo/publisher/headerbidding/MoPubHeaderBiddingTest;

    invoke-direct {p0, p2}, Lcom/mopub/mobileads/MoPubView;-><init>(Landroid/content/Context;)V

    return-void
.end method

```

The class is `MoPubHeaderBiddingTest$1` (I guess `$2` for the other) and is a subclass of `MoPubView`

Reply:
Thanks, TIL.